mlflow/pytorch/_lightning_autolog.py

```diff
@@ -0,0 +1,478 @@
+from packaging.version import Version
+import logging
+import mlflow.pytorch
+import os
+import shutil
+import sys
+import tempfile
+import warnings
+
+from mlflow.ml_package_versions import _ML_PACKAGE_VERSIONS
+from mlflow.exceptions import MlflowException
+from mlflow.utils.autologging_utils import (
+    ExceptionSafeAbstractClass,
+    BatchMetricsLogger,
+    MlflowAutologgingQueueingClient,
+    get_autologging_config,
+)
+
+logging.basicConfig(level=logging.ERROR)
+MIN_REQ_VERSION = Version(_ML_PACKAGE_VERSIONS["pytorch-lightning"]["autologging"]["minimum"])
+MAX_REQ_VERSION = Version(_ML_PACKAGE_VERSIONS["pytorch-lightning"]["autologging"]["maximum"])
+
+try:
+    import pytorch_lightning as pl
+    from pytorch_lightning.utilities import rank_zero_only
+
+    HAVE_LIGHTNING = True
+
+
+except ImportError:
+    HAVE_LIGHTNING = False
+
+    class pl:
+        Callback = object
+        __version__ = "0.0.0"
+
+    def rank_zero_only(f):
+        return f
```

Comment:
I think this is error-prone and confusing. Is there a way to avoid doing this?

Reply:
should be better now, wdyt